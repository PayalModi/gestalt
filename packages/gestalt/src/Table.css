.table {
  border-collapse: separate;
  border-spacing: 0 4px;
  cursor: default;
  width: 100%;
}

.th {
  composes: borderBottom from "./Borders.css";
  composes: paddingY3 from "./boxWhitespace.css";
  composes: paddingX3 from "./boxWhitespace.css";
}

.td {
  composes: paddingY3 from "./boxWhitespace.css";
  composes: paddingX3 from "./boxWhitespace.css";
}

.trClickable:hover {
  background-color: #efefef;
  cursor: pointer;
}

.trClickable:hover td:first-child {
  border-radius: 8px 0 0 8px;
}

.trClickable:hover td:last-child {
  border-radius: 0 8px 8px 0;
}
