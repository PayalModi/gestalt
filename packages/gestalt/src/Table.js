// @flow
import * as React from 'react';
import styles from './Table.css';

const TableBody = ({ children }: { children: React.Node }) => (
  <tbody>{children}</tbody>
);

const TableCell = ({
  children,
  colSpan,
  rowSpan,
}: {
  children: React.Node,
  colSpan?: number,
  rowSpan?: number,
}) => {
  let props = { className: styles.td };
  if (colSpan) {
    props = { ...props, colSpan };
  }
  if (rowSpan) {
    props = { ...props, rowSpan };
  }

  return <td {...props}>{children}</td>;
};

const TableHeader = ({ children }: { children: React.Node }) => (
  <thead className={styles.thead}>{children}</thead>
);

const TableHeaderCell = ({
  children,
  colSpan,
  rowSpan,
  scope = 'col',
}: {
  children: React.Node,
  colSpan?: number,
  rowSpan?: number,
  scope?: 'col' | 'colgroup' | 'row' | 'rowgroup',
}) => {
  let props = { className: styles.th, scope };
  if (colSpan) {
    props = { ...props, colSpan };
  }
  if (rowSpan) {
    props = { ...props, rowSpan };
  }

  return <th {...props}>{children}</th>;
};

const TableRow = ({ children }: { children: React.Node }) => (
  <tr>{children}</tr>
);

type TableProps = {|
  children: React.Node,
|};

export default class Table extends React.PureComponent<TableProps> {
  static Body = TableBody;

  static Cell = TableCell;

  static Header = TableHeader;

  static HeaderCell = TableHeaderCell;

  static Row = TableRow;

  render() {
    return (
      <div style={{ overflowX: 'scroll' }}>
        <table className={styles.table}>{this.props.children}</table>
      </div>
    );
  }
}
