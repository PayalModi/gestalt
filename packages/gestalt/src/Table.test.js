// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import Table from './Table.js';

describe('<Table />', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <Table>
          <div>rest of table</div>
        </Table>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('<Table.Body />', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <Table.Body>
          <div>rows</div>
        </Table.Body>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('<Table.Cell />', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(<Table.Cell>cell content</Table.Cell>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with colSpan and rowSpan', () => {
    const tree = renderer
      .create(
        <Table.Cell rowSpan={2} colSpan={3}>
          cell content
        </Table.Cell>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('<Table.Header />', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <Table.Header>
          <div>row with column names</div>
        </Table.Header>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('<Table.HeaderCell />', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(<Table.HeaderCell>column name</Table.HeaderCell>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with props', () => {
    const tree = renderer
      .create(
        <Table.HeaderCell rowSpan={2} colSpan={3} scope="row">
          row name
        </Table.HeaderCell>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('<Table.Row />', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <Table.Row>
          <div>row cells</div>
        </Table.Row>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
