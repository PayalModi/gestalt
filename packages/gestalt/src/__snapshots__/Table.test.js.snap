// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="box overflowScrollX paddingX1"
>
  <table
    className="table"
  >
    <div>
      rest of table
    </div>
  </table>
</div>
`;

exports[`renders correctly with border 1`] = `
<div
  className="borderColorLightGray box overflowScrollX paddingX1 rounding1 sizeSm solid"
>
  <table
    className="table"
  >
    <div>
      rest of table
    </div>
  </table>
</div>
`;
